.github/workflows/gh-pages.yml

```diff
@@ -0,0 +1,34 @@
+name: Build and Deploy Docs
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
```

Comment:
This goes out, was only for testing.

Reply:
It's nice to build docs, whenever someone sends a PR. That's why I left it.

Reply:
Indeed but then the name of the workflow is confusing and assumes deployment is also done.

Reply:
As explained below

* main => Build + Deploy
* PR => Build only

So I think the current name is fair enough, since the trigger from `main` is going to be the most common